metadata/src/test/java/org/apache/kafka/metalog/LocalLogManager.java

```diff
@@ -732,12 +742,22 @@ public long prepareAppend(
         if (throwOnNextAppend.getAndSet(false)) {
             throw new BufferAllocationException("Test asked to fail the next prepareAppend");
         }
-
```

Comment:
Should we also need to check current node is lead here?
(https://github.com/apache/kafka/blob/0f7cd4dcdeb2c705c01743927e36b66b06010f20/raft/src/main/java/org/apache/kafka/raft/RaftClient.java#L150

Reply:
Thanks!